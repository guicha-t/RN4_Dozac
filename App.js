import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';

import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import Router from './src/Router';

import { watcherSaga } from './src/saga/request';

import rootReducer from './src/store/reducers';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose(applyMiddleware(sagaMiddleware)));

sagaMiddleware.run(watcherSaga);

export default function App() {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Router />
      </NavigationContainer>
    </Provider>
  );
}
