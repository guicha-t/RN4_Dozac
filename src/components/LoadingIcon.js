import * as React from 'react';
import { ActivityIndicator, View } from 'react-native';

export default function LoadingIcon() {
  return (
    <View style={{ flex: 1, justifyContent: 'center' }}>
      <ActivityIndicator size="large" color="#0000ff" />
    </View>
  );
}
