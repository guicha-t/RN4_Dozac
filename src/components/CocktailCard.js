/* eslint-disable no-unused-vars */
import React from 'react';

import { View, Text, Image, StyleSheet } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';

import LoadingIcon from './LoadingIcon';

const styles = StyleSheet.create({});

export default function CocktailCard({ name, pic, cocktail, onPress, sty }) {
  if (cocktail === undefined) {
    return <LoadingIcon />;
  }
  return (
    <View style={sty}>
      <TouchableOpacity
        style={{ justifyContent: 'center', alignItems: 'center' }}
        onPress={onPress}
      >
        <Image
          style={{ height: 70, width: 70, margin: 3 }}
          source={
            pic
              ? {
                  uri: `${pic}/preview`,
                }
              : require('../../assets/broken.png')
          }
          resizeMode="contain"
        />
        <View>
          <Text style={{ flexWrap: 'wrap', textAlign: 'center', color: 'brown' }}>{name}</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
