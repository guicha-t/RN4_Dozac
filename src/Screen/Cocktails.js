/* eslint-disable no-shadow */
import React, { useState, useEffect } from 'react';
import { View, TouchableOpacity, Text, StyleSheet, FlatList } from 'react-native';

import StyleWrapper from '../HOC/styleHOC';
import LoadingIcon from '../components/LoadingIcon';
import CocktailCard from '../components/CocktailCard';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'flex-start',
  },
  card: {
    width: '31%',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 3,
    borderRadius: 6,
    backgroundColor: 'blanchedalmond',
  },
});

const urlFamous = 'https://www.thecocktaildb.com/api/json/v2/9973533/popular.php';
const urlRandom = 'https://www.thecocktaildb.com/api/json/v2/9973533/random.php';

function Cocktails({ navigation }) {
  const [cocktails, setCocktails] = useState([]);
  const [loading, setLoading] = useState(true);

  const fetchCocktails = async url => {
    const res = await fetch(url || urlFamous);
    await res.json().then(res => setCocktails(res.drinks));
    setLoading(false);
  };

  useEffect(() => {
    fetchCocktails();
  }, [false]);

  if (loading) {
    return <LoadingIcon />;
  }

  return (
    <View style={{ flex: 1 }}>
      <View
        style={{
          justifyContent: 'center',
          alignItems: 'center',
          backgroundColor: 'blanchedalmond',
        }}
      >
        <Text style={{ color: 'brown', fontSize: 20 }}>Cocktails</Text>
      </View>

      <View style={{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }}>
        <TouchableOpacity
          style={{
            margin: 3,
            padding: 5,
            backgroundColor: 'blanchedalmond',
            borderWidth: 1,
            borderRadius: 6,
          }}
          onPress={() => {
            fetchCocktails(urlFamous);
          }}
        >
          <Text style={{ color: 'brown', margin: 3, marginLeft: 10, marginRight: 10 }}>
            les 20 les plus populaires
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={{
            margin: 3,
            padding: 5,
            backgroundColor: 'blanchedalmond',
            borderWidth: 1,
            borderRadius: 6,
          }}
          onPress={() => {
            fetchCocktails(urlRandom);
          }}
        >
          <Text style={{ color: 'brown', margin: 3, marginLeft: 10, marginRight: 10 }}>
            Al√©atoire
          </Text>
        </TouchableOpacity>
      </View>

      <FlatList
        numColumns={3}
        data={cocktails}
        scrollEnabled
        renderItem={({ item }) => (
          <CocktailCard
            sty={styles.card}
            name={item.strDrink}
            pic={item.strDrinkThumb}
            cocktail={item}
            onPress={() => {
              navigation.navigate('CocktailPage', { cocktailId: item });
            }}
          />
        )}
        contentContainerStyle={{
          justifyContent: 'space-between',
          alignItems: 'stretch',
        }}
        keyExtractor={(item, index) => `id${index}`}
      />
    </View>
  );
}

export default StyleWrapper(Cocktails);
