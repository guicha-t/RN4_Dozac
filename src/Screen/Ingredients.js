import React from 'react';
import { View, TouchableOpacity, Text, Image, Dimensions, StyleSheet, FlatList, TextInput } from 'react-native';

import StyleWrapper from '../HOC/styleHOC';
import LoadingIcon from '../components/LoadingIcon';
import { TranslateIngredients } from '../components/translation';
import CocktailCard from '../components/CocktailCard'

import { useState, useEffect } from 'react';
import axios from 'axios';
import { ScrollView } from 'react-native-gesture-handler';

const styles = StyleSheet.create({
  container: { 
    flex: 1, 
    flexDirection: 'row', 
    flexWrap: 'wrap',
    alignItems: 'flex-start',
  },
  card: {
    justifyContent: 'center', 
    alignItems: 'center', 
    margin: 3, 
    borderRadius: 6, 
    backgroundColor: 'blanchedalmond',
  }
})

function Ingredients({ navigation }) {
  const [cocktails, setCocktails] = useState([]);
  const [loading, setLoading] = useState(true);
  const [hasErrors, setErrors] = useState(false);
  const [ingredients, setIngredients] = useState([])

  const [name, setName] = React.useState('');

  const fetchIngredients = async (ingredients, name) => {
    const res = await fetch(urlArg(ingredients, name));
    const data = await res
    .json()
    .then(
      res => setCocktails(res.drinks)
    );
    setLoading(false);
  }
  
  function addIngredient(item) {
    setIngredients([...ingredients, item]);
    setName('');
  }

  function erraseIngredient(item) {
    const index = ingredients.indexOf(item);
    setIngredients(ingredients.splice(index, 1))
    fetchIngredients(ingredients, name)
  }

  const urlArg = (ingredients, id) => { 
    let res = 'https://www.thecocktaildb.com/api/json/v2/9973533/filter.php?i='
    for (let i = 0; i < ingredients.length; i++) {
      if (i + 1 === ingredients.length) {
        res += ingredients[i]
      } else {
        res += ingredients[i] + ','
      }
    }
    if (id.length != 0)
      res += ',' + id;
    console.log(res);
    return res;
  }

  useEffect(() => {
    fetchIngredients(ingredients, name);
  }, [name, ingredients]);

  if (loading) {
    return <LoadingIcon />;
  } else {
    console.log(cocktails, hasErrors);
  }

  return (
    <View style={{flex: 1}}>
        <View style={{ justifyContent: 'center', alignItems: 'center', backgroundColor: 'blanchedalmond'}}>
          <Text style={{color: 'brown', fontSize: 20}} >Ingredients</Text>
        </View>
        <View style={{margin: 10, paddingBottom: 20}}>
            <Text style={{fontSize: 18}}>Nom de votre ingredient</Text>
            <View style={{flexDirection: 'row'}}>
            <TouchableOpacity 
              style={{margin: 3, padding: 5, backgroundColor: 'blanchedalmond', borderWidth: 1, borderRadius: 6}}
              onPress={() => { addIngredient(name) }}
              >
                <Text>Ajouter</Text>
              </TouchableOpacity>
              <TextInput
              style={{ height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10, backgroundColor:'blanchedalmond'}}
              onChangeText={(text) => setName(text)}
              value={name}
            />
            </View>
        </View>
      <ScrollView>
      <View style={{flexDirection: 'row'}}>
        <View style={{flexDirection: 'row', backgroundColor: 'chocolate', margin: 3, minWidth: 60}}>
          {
            ingredients.map((item, index) => {
              return (
                <TouchableOpacity key={index} style={{flexWrap: 'wrap', maxHeight: 60, borderRadius: 6, backgroundColor: 'blanchedalmond', margin: 6, padding: 2}}
                  onPress={() => { erraseIngredient(item)}}>
                  <Text style={{margin: 5}}>{item}</Text>
                </TouchableOpacity>
              );
            })
          }
        </View>
        <View style={{flexDirection: 'row', backgroundColor: 'chocolate', margin: 3}}>
          <FlatList
            numColumns={2}
            data={cocktails}
            scrollEnabled={true}
            renderItem={({ item }) => 
              <CocktailCard
                sty={styles.card}
                name={item.strDrink}
                pic={item.strDrinkThumb}
                cocktail={item} 
                onPress={() => {
                  navigation.navigate('CocktailPage', {cocktailId: item})
                }} 
              />}
            contentContainerStyle={{
              justifyContent: 'flex-end',
            }}
            keyExtractor={(item, index) => 'id' + index}
          />
        </View>
      </View>
      </ScrollView>
    </View>
  );
}

export default (StyleWrapper(Ingredients))
