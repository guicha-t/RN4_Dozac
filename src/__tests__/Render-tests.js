/* eslint-disable no-undef */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CocktailCard from '../components/CocktailCard';
import LoadingIcon from '../components/LoadingIcon';
import Cocktails from '../screen/Cocktails';
import Ingredients from '../screen/Ingredients';
import App from '../../App';

it('CocktailCard - renders correctly', () => {
  const tree = renderer.create(<CocktailCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Cocktails - renders correctly', () => {
  const tree = renderer.create(<Cocktails />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Ingredients - renders correctly', () => {
  const tree = renderer.create(<Ingredients />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('LoadingIcon - renders correctly', () => {
  const tree = renderer.create(<LoadingIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('AppJs - renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
