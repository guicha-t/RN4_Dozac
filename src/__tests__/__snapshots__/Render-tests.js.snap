// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppJs - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column",
                          "justifyContent": "flex-start",
                        }
                      }
                    >
                      <ViewManagerAdapter_ExpoLinearGradient
                        proxiedProperties={
                          Object {
                            "colors": Array [
                              4293024281,
                              4294936141,
                              4294679921,
                            ],
                            "endPoint": undefined,
                            "locations": undefined,
                            "startPoint": undefined,
                          }
                        }
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "paddingTop": 30,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "column",
                                "justifyContent": "center",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flex": 0.4,
                                  "justifyContent": "flex-end",
                                }
                              }
                            >
                              <Image
                                source={1}
                                style={
                                  Object {
                                    "height": 240,
                                    "width": 240,
                                  }
                                }
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "flex": 0.4,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#DEDEDE",
                                    "borderRadius": 6,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                    "maxWidth": 400,
                                    "minHeight": 40,
                                    "opacity": 1,
                                    "paddingLeft": 10,
                                    "paddingRight": 10,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    theme={
                                      Object {
                                        "colors": Object {
                                          "disabled": "hsl(208, 8%, 90%)",
                                          "divider": "#bcbbc1",
                                          "error": "#ff190c",
                                          "grey0": "#393e42",
                                          "grey1": "#43484d",
                                          "grey2": "#5e6977",
                                          "grey3": "#86939e",
                                          "grey4": "#bdc6cf",
                                          "grey5": "#e1e8ee",
                                          "greyOutline": "#bbb",
                                          "platform": Object {
                                            "android": Object {
                                              "error": "#f44336",
                                              "primary": "#2196f3",
                                              "secondary": "#9C27B0",
                                              "success": "#4caf50",
                                              "warning": "#ffeb3b",
                                            },
                                            "ios": Object {
                                              "error": "#ff3b30",
                                              "primary": "#007aff",
                                              "secondary": "#5856d6",
                                              "success": "#4cd964",
                                              "warning": "#ffcc00",
                                            },
                                          },
                                          "primary": "#2089dc",
                                          "searchBg": "#303337",
                                          "secondary": "#8F0CE8",
                                          "success": "#52c41a",
                                          "warning": "#faad14",
                                        },
                                      }
                                    }
                                    underlayColor="white"
                                  >
                                    <Text />
                                  </View>
                                </View>
                                <Text
                                  style={
                                    Object {
                                      "color": "#152654",
                                      "fontSize": 20,
                                      "textAlign": "center",
                                    }
                                  }
                                  type="clear"
                                >
                                  Login
                                </Text>
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    theme={
                                      Object {
                                        "colors": Object {
                                          "disabled": "hsl(208, 8%, 90%)",
                                          "divider": "#bcbbc1",
                                          "error": "#ff190c",
                                          "grey0": "#393e42",
                                          "grey1": "#43484d",
                                          "grey2": "#5e6977",
                                          "grey3": "#86939e",
                                          "grey4": "#bdc6cf",
                                          "grey5": "#e1e8ee",
                                          "greyOutline": "#bbb",
                                          "platform": Object {
                                            "android": Object {
                                              "error": "#f44336",
                                              "primary": "#2196f3",
                                              "secondary": "#9C27B0",
                                              "success": "#4caf50",
                                              "warning": "#ffeb3b",
                                            },
                                            "ios": Object {
                                              "error": "#ff3b30",
                                              "primary": "#007aff",
                                              "secondary": "#5856d6",
                                              "success": "#4cd964",
                                              "warning": "#ffcc00",
                                            },
                                          },
                                          "primary": "#2089dc",
                                          "searchBg": "#303337",
                                          "secondary": "#8F0CE8",
                                          "success": "#52c41a",
                                          "warning": "#faad14",
                                        },
                                      }
                                    }
                                    underlayColor="white"
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                              <View
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#DEDEDE",
                                    "borderRadius": 6,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "margin": 5,
                                    "maxWidth": 400,
                                    "minHeight": 40,
                                    "opacity": 1,
                                    "paddingLeft": 10,
                                    "paddingRight": 10,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    theme={
                                      Object {
                                        "colors": Object {
                                          "disabled": "hsl(208, 8%, 90%)",
                                          "divider": "#bcbbc1",
                                          "error": "#ff190c",
                                          "grey0": "#393e42",
                                          "grey1": "#43484d",
                                          "grey2": "#5e6977",
                                          "grey3": "#86939e",
                                          "grey4": "#bdc6cf",
                                          "grey5": "#e1e8ee",
                                          "greyOutline": "#bbb",
                                          "platform": Object {
                                            "android": Object {
                                              "error": "#f44336",
                                              "primary": "#2196f3",
                                              "secondary": "#9C27B0",
                                              "success": "#4caf50",
                                              "warning": "#ffeb3b",
                                            },
                                            "ios": Object {
                                              "error": "#ff3b30",
                                              "primary": "#007aff",
                                              "secondary": "#5856d6",
                                              "success": "#4cd964",
                                              "warning": "#ffcc00",
                                            },
                                          },
                                          "primary": "#2089dc",
                                          "searchBg": "#303337",
                                          "secondary": "#8F0CE8",
                                          "success": "#52c41a",
                                          "warning": "#faad14",
                                        },
                                      }
                                    }
                                    underlayColor="white"
                                  >
                                    <Text />
                                  </View>
                                </View>
                                <Text
                                  style={
                                    Object {
                                      "color": "#152654",
                                      "fontSize": 20,
                                      "textAlign": "center",
                                    }
                                  }
                                  type="clear"
                                >
                                  Sign In
                                </Text>
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                    theme={
                                      Object {
                                        "colors": Object {
                                          "disabled": "hsl(208, 8%, 90%)",
                                          "divider": "#bcbbc1",
                                          "error": "#ff190c",
                                          "grey0": "#393e42",
                                          "grey1": "#43484d",
                                          "grey2": "#5e6977",
                                          "grey3": "#86939e",
                                          "grey4": "#bdc6cf",
                                          "grey5": "#e1e8ee",
                                          "greyOutline": "#bbb",
                                          "platform": Object {
                                            "android": Object {
                                              "error": "#f44336",
                                              "primary": "#2196f3",
                                              "secondary": "#9C27B0",
                                              "success": "#4caf50",
                                              "warning": "#ffeb3b",
                                            },
                                            "ios": Object {
                                              "error": "#ff3b30",
                                              "primary": "#007aff",
                                              "secondary": "#5856d6",
                                              "success": "#4cd964",
                                              "warning": "#ffcc00",
                                            },
                                          },
                                          "primary": "#2089dc",
                                          "searchBg": "#303337",
                                          "secondary": "#8F0CE8",
                                          "success": "#52c41a",
                                          "warning": "#faad14",
                                        },
                                      }
                                    }
                                    underlayColor="white"
                                  >
                                    <Text />
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flex": 0.2,
                                  "justifyContent": "flex-end",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#696969",
                                    "fontSize": 8,
                                  }
                                }
                              >
                                Romain GADRAT - Thomas GUICHARD - Artem VATOUTINE
                              </Text>
                            </View>
                          </View>
                        </View>
                      </ViewManagerAdapter_ExpoLinearGradient>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      />
    </View>
  </RNCSafeAreaView>
</View>
`;

exports[`CocktailCard - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`LoadingIcon - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
